Assets/Scripts/Player/PlayerDataManager.cs

```diff
@@ -7,9 +7,21 @@ public class PlayerDataManager : MonoBehaviour
     {
         public PlayerData PlayerData { get; set; }
 
-        private void Awake()
+        private void Start()
         {
             PlayerData = new PlayerData(gameObject.transform.position);
         }
+        
+        /*
+         * Since the rigidbody is not directly on the game object that is moving. We cannot use it to get the velocity.
+         * The velocity needs to be computed manually.
+         */
+        private void Update()
```

Comment:
Might not be needed later since I found a fix to having the rigid body attached to the player directly in my branch, but it works for now, so we should refactor this later

Reply:
Yup, thats why I wanted your PR to be merged so I can adapt my code.